grid-client/deployer/deployer.go

```diff
@@ -448,7 +447,7 @@ func (d *Deployer) BatchDeploy(ctx context.Context, deployments map[uint32][]gri
 	}
 
 	contracts, index, err := d.substrateConn.BatchCreateContract(d.identity, contractsData)
-	if err != nil && index == nil {
+	if err != nil {
```

Comment:
Removing this check removes the whole purpose of this method as non-atomic deployer. Because if index point to a contract 5 for example that means 4 contracts were created but there deployments won't be created. which other deployers are using it based on that (some contracts and deployments will be created and other will fail). also it won't cancel the contracts created so it will leave some contracts behind without the user knowing.